great_expectations/dataset/util.py

```diff
@@ -115,11 +116,21 @@ def recursively_convert_to_json_serializable(test_obj):
 
     """
     # Validate that all aruguments are of approved types, coerce if it's easy, else exception
+    # print(type(test_obj), test_obj)
+    #Note: Not 100% sure I've resolved this correctly...
+    try:
+        if np.isnan(test_obj):
+            return None
+    except TypeError:
+        pass
+    except ValueError:
+        pass
+
     if isinstance(test_obj, (string_types, integer_types, float, bool)):
         # No problem to encode json
         return test_obj
 
-    elif test_obj is None:
+    if isinstance(test_obj, (str, int, float, bool)):
```

Comment:
This seems wrong...it's redundant with the check above.

Reply:
Good catch.